src/Sylius/Behat/Context/Setup/PaymentMethodContext.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Setup;
+
+use Behat\Behat\Context\Context;
+use Sylius\Bundle\CoreBundle\Test\Services\PaymentMethodNameToGatewayConverterInterface;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Payment\Model\PaymentMethodInterface;
+use Sylius\Component\Payment\Repository\PaymentMethodRepositoryInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+final class PaymentMethodContext implements Context
+{
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @var FactoryInterface
+     */
+    private $paymentMethodFactory;
+
+    /**
+     * @var PaymentMethodNameToGatewayConverterInterface
+     */
+    private $paymentMethodNameToGatewayConverter;
+
+    /**
+     * @var PaymentMethodRepositoryInterface
+     */
+    private $paymentMethodRepository;
+
+    /**
+     * @param SharedStorageInterface $sharedStorage
+     * @param FactoryInterface $paymentMethodFactory
+     * @param PaymentMethodNameToGatewayConverterInterface $paymentMethodNameToGatewayConverter
+     * @param PaymentMethodRepositoryInterface $paymentMethodRepository
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        FactoryInterface $paymentMethodFactory,
+        PaymentMethodNameToGatewayConverterInterface $paymentMethodNameToGatewayConverter,
+        PaymentMethodRepositoryInterface $paymentMethodRepository
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->paymentMethodFactory = $paymentMethodFactory;
+        $this->paymentMethodNameToGatewayConverter = $paymentMethodNameToGatewayConverter;
+        $this->paymentMethodRepository = $paymentMethodRepository;
+    }
+
+    /**
+     * @Given the store has a payment method :paymentMethodName with a code :paymentMethodCode
+     */
+    public function theStoreHasAPaymentMethodWithACode($paymentMethodName, $paymentMethodCode)
+    {
+        $paymentMethod = $this->createPaymentMethodFromNameAndCode($paymentMethodName, $paymentMethodCode);
+
+        $this->sharedStorage->set('payment_method', $paymentMethod);
+        $this->paymentMethodRepository->add($paymentMethod);
+    }
+
+    /**
+     * @Given /^the store has a (payment method "([^"]*)") disabled$/
```

Comment:
@pjedrzejewski not exactly, this way running `the store has a payment method "Ekopolex" with a code "EKOPOLEX"` won't throw any error, but `the store has a payment method "Ekopolex" disabled` run without the previous one will.

What about `the payment method "Ekopolex" is disabled` or `that payment method is disabled`?
It will allow for a scenario like:
```cucumber
Scenario: ...
    Given the store has a payment method "Ekopolex" with code "EKOPOLEX"

    But that payment method is disabled # 1.
    But it is disabled # 2.
```

If we want to achieve that in one step, let's use `the store has a disabled payment method "Ekopolex"` which creates a disabled payment method :)